kernel_crawler/ubuntu.py

```diff
@@ -17,19 +17,20 @@ def to_driverkit_config(self, release, deps):
         dk_configs = {}
         krel, kver = release.split("/")
         for dep in deps:
-            if dep.find("headers") != -1:
+            if 'headers' in dep:
+
+                # set a default flavor
+                flavor = 'generic'
+                # capture the flavor from the string after 'linux-' in the url subdir
                 # example: http://security.ubuntu.com/ubuntu/pool/main/l/linux-oracle/linux-headers-4.15.0-1087-oracle_4.15.0-1087.95_amd64.deb
-                d = re.search(r"(\bl/linux(\-.+\/)?\b)", dep)
-                if d is None:
-                    continue
+                flavor_capture = re.search(r"^.*l/linux-(.+)/.*$", dep)
```

Comment:
Is this an improved version? I mean, shouldn't the old regex work also? It was working before!

Reply:
The previous regex included the `-` in the capture group and the ending `/`. To get around that problem, the code also previously was doing this to get the kernel release:
```python
                d = d.group(0)
                 target = "ubuntu" + d[7:len(d) - 1]
                 release = krel + d[7:len(d) - 1]
```
...which is a bit hard to follow in code (where did that 7 even come from?)

So I simplified it to exclude the `-` and the `/` from the capture group and actually just capture the flavor directly. Then we do not need to use array slicing to reliably get the flavor.

The previous regex also used: `(\b` and `\b)` to start and end the expression, which I just didn't recognize. Using `^.*` and `.*$`is more generic and explicitly defines the start/end of the expression.

Reply:
Uh thank you! Yep it looks really better eheh
thanks for the kind explanation!

Reply:
Good catch @EXONER4TED! thank you. We could also leave the word boundaries, with something like `\bl/linux-(.+)/\b` as we don't need to catch start and end matches.

LGTM, I don't get how it worked before :/